Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionResultPositiveOtherWarningFragment.kt

```diff
@@ -178,6 +188,34 @@ class SubmissionResultPositiveOtherWarningFragment : Fragment(),
     override fun onKeySharePermissionGranted(keys: List<TemporaryExposureKey>) {
         super.onKeySharePermissionGranted(keys)
         submissionViewModel.submitDiagnosisKeys(keys)
+        if (submissionViewModel.submissionState != ApiRequestState.SUCCESS) {
+            showSubmitErrorDialog()
+        }
```

Comment:
I think this would show the "no diagnosis key" dialog for all sorts of failures and not only when no keys are available.

Also, error handling is already in onViewCreated -> submissionViewModel.submissionError.observeEvent. I think we should avoid having it at two places.

Reply:
Made the necessary changes